### PR TITLE
Update transaction model and UI to allow for multiple payees

### PR DESCRIPTION
**Pull request by Steven-Jia**

### CODE REVIEW

File: src/main/java/seedu/address/model/transaction/Transaction.java [hunk 1]
```diff
@@ -1,20 +1,32 @@
 package seedu.address.model.transaction;
 
+import java.util.Date;
+import java.util.Objects;
+import java.util.Set;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
```
Comment (phmignot) [inline src/main/java/seedu/address/model/transaction/Transaction.java]: From Travis with love : 
"Wrong lexicographical order for 'java.time.Clock' import. Should be before 'java.util.Set'.
Wrong lexicographical order for 'java.time.Instant' import. Should be before 'java.util.Set'. 
Wrong lexicographical order for 'java.time.ZoneId' import. Should be before 'java.util.Set'. "

File: src/main/java/seedu/address/model/transaction/Transaction.java [hunk 2]
```diff
@@ -64,21 +86,27 @@ public boolean equals(Object other) {
         }
 
         Transaction otherTransaction = (Transaction) other;
-        return otherTransaction.getPayer().equals(this.getPayer())
+        return otherTransaction.getId().equals(this.getId())
```
Comment (phmignot) [inline src/main/java/seedu/address/model/transaction/Transaction.java]: By definition, getId().equals(this.getId()) should be enough for the method equals ?

File: src/main/java/seedu/address/model/transaction/Transaction.java [hunk 3]
```diff
@@ -23,20 +35,30 @@ public Transaction(Person payer, Double amount, String description, UniquePerson
...
-        for(Person p: payeesToAdd) {
+        for (Person p: payeesToAdd) {
             try {
                 payees.add(p);
             } catch (DuplicatePersonException e) {
                 System.out.println("Duplicate person" + p.getName() + " not added to list of payees");
             }
         }
 
+        this.id = 0; // TODO: Ensure that this increments by 1 for each new transaction
```
Comment (phmignot) [inline src/main/java/seedu/address/model/transaction/Transaction.java]: Just add in the class : 
private static int lastTransactionId = 0;

and in the constructor : 
this.id = lastTransactionId;
lastTransactionId++;

File: src/main/java/seedu/address/model/transaction/Transaction.java (hunk 2 above)
Comment (Steven-Jia) [reply src/main/java/seedu/address/model/transaction/Transaction.java]: Thanks, I'll change that

**Comment by phmignot** [comment]
I'm okay with your change. Just have to solve conflict when merging

**Comment by Steven-Jia** [comment]
Will wait till after this week's tutorial since we still need to implement retrieving a Transaction by name

**Pull request merged**